src/coprocessor/codec/mysql/time.rs

```diff
@@ -311,6 +333,41 @@ impl Time {
         Time::new(t, tp, fsp as i8)
     }
 
+    pub fn from_duration(tz: &FixedOffset, d: &MyDuration) -> Result<Time> {
+        let local = Local::now();
+        let t = try!(ymd_hms_nanos(
```

Comment:
I don't think it's correct to fill the time with local time and some arbitrary timezone.